src/tinc_global.c

```diff
@@ -0,0 +1,18 @@
+#include "system.h"
+
```

Comment:
This should also include `"tinc_global.h"`. I think the `tinc_` prefix for the filename is also not necessary. Last but not least, it might be nice to ensure the .h and .c file declare/define the exact same variables.

Reply:
The whole idea of rearranging globals was a mistake on my part. It's probably best to do this separately later.

I created simple wrappers around `main()` to avoid linker errors. This should suffice and is more straightforward.

There are some intermittent failures with really weird [errors](https://github.com/hg/tinc/runs/5523326249?check_suite_focus=true#step:11:1002). Not sure what's causing it, as it only shows in github CI. I'll look into it.